readers/common/src/main/java/com/criteo/hadoop/garmadon/reader/metrics/ConsumerMetrics.java

```diff
@@ -0,0 +1,5 @@
+package com.criteo.hadoop.garmadon.reader.metrics;
+
+public interface ConsumerMetrics {
+    void stop();
```

Comment:
I find that this interface is not clearly obvious and is only the result of using 2 implementations. In other words, just having a close method seems to not say much about ConsumerMetrics. I would have expected such an interface to also have a start method because starting everything via statics is only a preference of PrometheusHttpConsumerMetrics
Maybe too early to extract an interface. Another way to approach this in terms of design can be to have a DefaultConsumerMetrics, that do nothing and the PrometheusHttpConsumerMetrics would extend this default class

Reply:
Don't see the issue or whet would be improved with DefaultConsumerMetrics instead of NoneConsumerMetrics (I can rename it if you want)
I will add start method to make it more "generic"

Reply:
The problem is not the name. The way the interface looked just shows that the reader class has very little control over ConsumerMetrics (just stopping it and this is only usefull for one of the implementations), and indeed, I think that there is no need to add the ConsumerMetrics to the reader class. If we want to properly stop Prometheus metrics we can register to the CompletableFuture returned by the stop() method of the reader and close the Prometheus stuff with whenComplete.